### ISSUE TITLE
Don't rebuild speex

### ISSUE THREAD

**Issue by Ndolam** [Thorn-Inurcide/tde-slackbuilds#15]
There is a speex slackbuild at https://slackbuilds.org/.
If speex is already installed when BUILD-TDE.sh is run, I don't think BUILD_TDE.sh should replace the existing speex package with its own.

**Comment by Ndolam** [comment]
Also, TDE's mp4v2 collides with Slackbuild's libmp4v2.  This could be more of a problem than speex, since the Slackbuild package has /usr/lib64/libmp4v2.a, whereas the TDE package doesn't have that.

**Comment by Thorn-Inurcide** [comment]
I wonder why they put lib in front of the actual library..and is that a
static library??

Hmm I'll have to look at that tomorrow after work I consider that a serious
issue to look over

**Comment by Thorn-Inurcide** [comment]
They don't have static disabled

**Comment by Ndolam** [comment]
I don't know how many people in the world will want to statically include libspeex, but you can bet there is someone out there...

**Comment by Thorn-Inurcide** [comment]
Lol I'll check these things out tomorrow I'm getting ready to lay down
soon. Thanks again

**Comment by Thorn-Inurcide** [comment]
@Ndolam  I will be doing a fresh install in a little bit and go over these issues.
I decided I'm not conforming my mpv42 SlackBuild to SBo's
But I will include a check for theirs like we talked about as well as with speex and the others.

**Comment by Thorn-Inurcide** [comment]
@Ndolam  I will be doing a fresh install in a little bit and go over these issues.
I decided I'm not conforming my mpv42 SlackBuild to SBo's
But I will include a check for theirs like we talked about as well as with speex and the others.

**Comment by Ndolam** [comment]
OK.  Just out of curiosity, do you feel the SBo package has shortcomings?

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
I don't see any need to have lib prefixed in front of the name, nor any reason to build static libraries.
If anyone wants those things though,the SBo build is there. I'll have the script check for libmp4v2 as well as just mp4v2 to prevent a rebuild if either are present. So if someone does want the SBo build and/or the static libs, they can keep them and this won't alter anything on their machines. As far as shortcomings no I see none, but I also see no reason to change mine. Nothing against SBo, i use some scripts from there and have some submitted to them, but these aren't for SBo so I'm not too worried about their standards. If I was going to submit something them I would care though and take all of their submission guidelines and other peoples builds into consideration. But I can easily just preform a check for the package and skip build mp4v2 if the user has libmp4v2 from SBo.

**Comment by Ndolam** [comment]
Assuming you mean the name of the package when you say "I don't see any need to have lib prefixed in front of the name", I agree with you.  Reading the comment at the top of /var/log/packages/libmp4v2-2.0.0-x86_64-1_SBo it would appear the comment may refer to a older version where they only had the library.  I just looked at the Slackbuilds on my system whose names start with "lib", and a lot have no files in /usr/bin or /bin or /sbin or /usr/sbin, but a number of them do.  Perhaps their naming policy is "if it has a library file, the package name starts with 'lib'".  Ho hum.

**Comment by Thorn-Inurcide** [comment]
it appears Arch Linux and Void Linux have it prefixed with lib also. So I'm guessing the other distros are doing it too.
I guess it will be painless for me to rename it to libmp4v2. But we will still do the check anyways to prevent a rebuild

**Comment by Thorn-Inurcide** [comment]
@Ndolam 

So I was thinking, If I add checks for these packages that remove them if they are already on the system, this might be a problem for the users who want them rebuilt and overwritten with what ones they previously had. The user has to manually select these packages to build them. Do you think maybe we should just leave this and hope the user knows the packages on the system and just doesn't select them if they don't want them rebuilt?

Also mp4v2 has been renamed to libmp4v2

**Comment by Thorn-Inurcide** [comment]
@ryanpcmcquen 
@Ray-V 

opinions here?

**Comment by ryanpcmcquen** [comment]
I would say a note in the README would suffice to check against your own packages.

**Comment by Ndolam** [comment]
One approach is to say "the user should know all the packages on his/her system and should read all the READMEs".  It is hard to argue against this being the responsible thing to do.  However, people being people, it might be nice for the build script to do a little hand-holding.

For example, in principle BUILD-TDE.sh could annotate the relevant lines with "you already have a package with this name" (or "you already have a package which provides these files" or ...) and that might remind the person building TDE that they might need to think about that for a few seconds.

**Comment by Thorn-Inurcide** [comment]
@Ndolam 

For example, in principle BUILD-TDE.sh could annotate the relevant lines with "you already have a package with this name" (or "you already have a package which provides these files" or ...) and that might remind the person building TDE that they might need to think about that for a few seconds.

This is nice, but I'm not sure how I can make the script check for the package during the package selection stage. I'm sure it is possible, I just would need to think about it and figure it out. perhaps check for them in advance then if they are there could display the reminder during the selection stage.
let me check this out, because I don't mind doing that. I just don't want to block out people from being abale to rebuild a dep if they want just because it is already on the system. I'm someone who rebuilds everything when I re-run this or any of my other scripts I have that are unrelated to TDE

**Comment by Thorn-Inurcide** [comment]
I added a note in the README for the time being

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
Adding some my SlackBuilds to TDE
What was it you said you wanted added for support? I remember you mentioning some things in another issue

**Comment by Ndolam** [comment]
At this point I now have only a  vague recollection of what the issue is.
I'm not immediately sure why I can't find anything.
However, FWIW, here is my guess...
My main use of TDE is korganizer, which requires kdepim.  However, I have this vague recollection that the build scripts (at the time) didn't compile all of the trinity packages, and one or more may have had some relationship to korganizer.
(For example, the other day I wanted to change the first day of the week from Monday to Sunday in korganizer, but I am supposed to do that through some control panel, which I either (a) could not find amongst anything I had on my system, or (b) could not cause it to run.  I probably need to d/l the latest version of the build scripts and run them again, and then I might figure out what I am missing.  (Somehow I managed to lose the build packages and (more curiously) /var/log/packages shows no trinity packages.)

**Comment by Thorn-Inurcide** [comment]
Hmm I'll hunt through the closed issues if I'm able to do that.  I'm pretty
sure there was a specific package you were requesting to add some kind of
support to something you use.  I'll look over that tomorrow. Getting ready
to sleep right now.

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
I finally found it

When I say "everything", I see that there aren't builds for everything. For example, I see there is no libcaldav. (One of my interests with TDE is korganizer, and having caldav support would be nice.) Can anyone tell me whether the things that are missing are missing because

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
Is that support still missing?

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
I'm guessing it is seeing how I just located the download for it

**Comment by Thorn-Inurcide** [comment]
@Ndolam 
libcaldav has been added, you want to rebuild TDE and check it out?

I don't know anything about korganizer so I have no idea what effect libcaldav does or doesn't have on it

Also added another theme and a twin theme

want to start adding as much as possible to this as I get time.

**Comment by Ndolam** [comment]
Thorn, sorry for the slow response, too many things on fire this week.

I will rebuild TDE asap and report back.

My understanding of all this is imperfect, but (AIUI) libcaldav allows korganizer to talk to calendar servers, rather than only using local calendar files on disk.  This makes things convenient if you need to share your calendar with others and/or if you want to access your calendar from multiple devices.  When I last tried this with korganizer (old version) it didn't work properly, but maybe things have improved.

I highly recommend korganizer, it is my favourite bit of TDE.

**Comment by Thorn-Inurcide** [comment]
Cool man, take your time if you're  busy no rush. I'm going to do
libcarddav and maybe the python support next

**Comment by Ndolam** [comment]
I downloaded tde-slackbuilds-master.zip about 2 hours ago and asked to build everything.
It ended up with 
+==============================================================================
| Installing new package /tmp/tdegwenview-i18n-R14.0.3-noarch-1_tde.txz
+==============================================================================

Verifying package tdegwenview-i18n-R14.0.3-noarch-1_tde.txz.
Installing package tdegwenview-i18n-R14.0.3-noarch-1_tde.txz:
PACKAGE DESCRIPTION:
# tdegwenview-i18n (Internationalization files for gwenview.)
#
# For languages:
#
# 
#
Package tdegwenview-i18n-R14.0.3-noarch-1_tde.txz installed.


Script started, file is /tmp/libmp4v2-build-log
Source 'mp4v2-2.0.0.tar.bz2' not available yet...
Will download file to /root/tde-slackbuilds-master/Misc/libmp4v2
http://mp4v2.googlecode.com/files/mp4v2-2.0.0.tar.bz2:
2016-09-28 20:23:11 ERROR 404: Not Found.
libmp4v2.SlackBuild FAILED at line 114
Script done, file is /tmp/libmp4v2-build-log
Cannot install /tmp/libmp4v2-2.0.0-*-1*.txz:  file not found

ls: cannot access '/var/log/packages/libmp4v2-*2.0.0-*-1*': No such file or directory

      Error:  libmp4v2 package build failed
      Check the build log /tmp/libmp4v2-build-log


I see it at https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/mp4v2/mp4v2-2.0.0.tar.bz2 but I'm not sure if that is the preferred place.

**Comment by Thorn-Inurcide** [comment]
Shit I forgot
That file is no longer hosted there
Grab the tarball from sbo and place it in your src folder I will update the
download link tomorrow

**Comment by Ndolam** [comment]
Done and compiled

**Comment by Thorn-Inurcide** [comment]
Lol awesome

**Comment by Ndolam** [comment]
k9crash is different:
checking ffmpeg/avcodec.h presence... no
checking for ffmpeg/avcodec.h... no
checking libavcodec/avcodec.h usability... no
checking libavcodec/avcodec.h presence... no
checking for libavcodec/avcodec.h... no
configure: error: libavcodec may be missing
k9copy.SlackBuild FAILED at line 87
Script done, file is /tmp/k9copy-build-log
Cannot install /tmp/k9copy-R14.0.3-*-1*.txz:  file not found

ls: cannot access '/var/log/packages/k9copy-*R14.0.3-*-1*': No such file or directory

      Error:  k9copy package build failed
      Check the build log /tmp/k9copy-build-log

This is in a newly-installed vm with no packages added.

I tried installing ffmpeg-3.1.1-x86_64-1alien and the k9copy build got further, but dies soon enough.

Which ffmpeg does this want/need?

**Comment by Thorn-Inurcide** [comment]
Umm wow
Why not build my ffmpeg? Select my slack builds repo there's an ffmpeg.sh
that builds it with almost every single dep all the source code is
downloaded for you but the binary is non redistributable so you have to
compile it yourself.  Just run the FFmpeg.sh and it will do everything.
Check the sh to see if libmp4v2 is in there if so dele that line since you
have it already. Delete the speex line too

**Comment by Ndolam** [comment]
Before you sent out your message I tried the usual slackbuild for ffmpeg.
k9copy compiled, so I guess ffmpeg 3.1.1 is too new for k9copy.
(And the following packages are happily compiling, at least down to tde-style=lipstik)

I'm looking at your ffmpeg build.  Does it really download all the source code?  I don't see that in the code of FFmpeg.sh

**Comment by Thorn-Inurcide** [comment]
Because that script goes into the slack build folder and each individual
script had the download in it

**Comment by Ndolam** [comment]
OK, now I see.
The tde "build of everything" just finished with lxml, so it looks like it is all good.  Now to test it, of course.

**Comment by Thorn-Inurcide** [comment]
Let me know how it goes

**Comment by Ndolam** [comment]
So far so good with all the basic stuff I have tried.
Next on the list will be to see if I can convince korganizer to talk to a caldav server I have set up.

**Comment by Thorn-Inurcide** [comment]
That's good to hear
I have to fix tdm later.
Getting ready to move again so I'll just be making note of it in known
issues for now

On Sep 29, 2016 8:17 PM, "Ndolam" <notifications@github.com> wrote:

So far so good with all the basic stuff I have tried.
Next on the list will be to see if I can convince korganizer to talk to a
caldav server I have set up.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Thorn-Inurcide/tde-slackbuilds/issues/15#issuecomment-250627819>,
or mute the thread
